Model/Order/Action/Decline.php

```diff
@@ -119,10 +119,17 @@ protected function refund($order)
      */
     protected function setOrderStatusDecline(Order $order)
     {
+        $orderState = $order->getHoldBeforeState();
+        $orderStatus = $order->getHoldBeforeStatus();
+        if ($orderState === Order::STATE_HOLDED && $orderStatus === OrderRis::STATUS_KOUNT_DECLINE) {
```

Comment:
Perhaps even more accurate way to handle the data corruption would be to check for State = Holded alone. Checking both state and status does prevent duplicate of identical calls, but would not protected from sequence of different calls, such as State = Review, State = Declined, followed by restore.